### ISSUE TITLE
Emit deprecation warning if two-line section titles are used

### ISSUE THREAD

**Issue by mojavelinux** [asciidoctor/asciidoctor#418]
If two-line section titles are used in a document, emit a deprecation warning by default. There is a consensus in the Asciidoctor community that the two-line section titles are confusing, harder to maintain and allow inconsistent formatting between documents and projects. To move AsciiDoc forward, Asciidoctor should treat the two-line section as deprecated.

As with other syntax deprecations being considered in Asciidoctor, the author should be able to control warnings using the deprecation attribute.

```
:deprecations: (enforce|warn|off)
```
- `enforce` will raise an error if a deprecated syntax is used (or simply ignore it)
- `warn` (the default) will emit a warning
- `off` will disable any deprecation checks.

**Comment by mojavelinux** [comment]
The deprecation warning should only show if `compat-mode` is not enabled.

**Comment by esiegerman** [comment]
-1

> I think [two-line section titles] hurts the AsciiDoc syntax, both from an adoption standpoint and a readability standpoint.

I'm brand-new to AsciiDoc, and not yet a member of the community, so have no opinion re. adoption.  As for readability, though, I strongly disagree.  That very feature is one of the things that drew me to AsciiDoc.  I find two-line headings far more readable than wiki-style one-line ones.  (Harder to type, to be sure, but easier to read.)

"AsciiDoc files are designed to be _viewed_, edited _and printed directly_ or translated to other presentation formats." -- Stuart Rackham (emphasis added)

In other words, easy editability was one of his goals, but by no means the only one; and in my opinion, the complementary goal, of having AsciiDoc files "consumable" as is, shouldn't be sacrificed to it.

[BTW, I imagine there's a better forum for discussing such general issues than one implementation's issue tracker.  Where might that be?]

Thanks for listening.

**Comment by mojavelinux** [comment]
> [BTW, I imagine there's a better forum for discussing such general design issues than one implementation's issue tracker. Where might that be?]

Asciidoctor list: http://discuss.asciidoctor.org/
AsciiDoc & AsciiDoc Python list: https://groups.google.com/forum/#!forum/asciidoc

But here is fine too. Most people involved are watching all these places.
